apps/passport-client/src/service-worker.ts

```diff
@@ -1,65 +1,307 @@
+ * where we want to optimize for speed and limit bandwidth at the cost of
+ * freshness.  This cache never has additional resources added.
+ * These resources are updated only when a new service worker is installed.
+ * since htat update isn't guaranteed to succeed, this list shouldn't contain
+ * anything which could cause compatibility problems if it isn't updated.
+ */
+const STABLE_CACHE_NAME = "v1-stable";
+const STABLE_CACHE_RESOURCES = [
+  "/favicon.ico",
+  "/semaphore-artifacts/16.wasm",
+  "/semaphore-artifacts/16.zkey",
+  "/artifacts/zk-eddsa-event-ticket-pcd/circuit.wasm",
+  "/artifacts/zk-eddsa-event-ticket-pcd/circuit.zkey",
+  "/fonts/IBMPlexSans-Regular.ttf",
+  "/fonts/IBMPlexSans-Medium.ttf",
+  "/fonts/IBMPlexSans-Light.ttf",
+  "/fonts/IBMPlexSans-ExtraLight.ttf",
+  "/fonts/IBMPlexSans-Regular.woff",
+  "/fonts/IBMPlexSans-Medium.woff",
+  "/fonts/IBMPlexSans-Light.woff",
+  "/fonts/IBMPlexSans-ExtraLight.woff"
+];
 
-  await Promise.all(
-    ["/", "/index.html", "/js/index.js"].map((item) => cache.delete(item))
-  );
-  await Promise.all(impermanentCache.map((item) => cache.delete(item)));
+/**
+ * Used to extract a key from a request, to compare it to the lists of
+ * cache resources above.  This may need updating if the list above ever
+ * includes more complex URLs, with parameters, or across domains.
+ */
+function requestToItemCacheKey(request: Request): string {
+  return new URL(request.url).pathname;
+}
 
-  await cache.addAll(resources);
+function logCacheName(cacheName: string): string {
+  return cacheName.substring(0, 9);
 }
 
-self.addEventListener("install", (event: any) => {
-  console.log(`[SERVICE_WORKER] installing ${process.env.SW_ID}`);
-  (self as any).skipWaiting();
+/**
+ * Helpers for quickly enabling or disabling verbose logging.
+ */
+const swLog = {
+  verbose: true, // TODO(artwyman): Disable before merging.
+  tag: `[SERVICE_WORKER][${process.env.SW_ID.substring(0, 4)}]`,
+  I: function (msg: string) {
+    console.log(this.tag, msg);
+  },
+  V: function (msg: string) {
+    if (this.verbose) {
+      console.debug(this.tag, msg);
+    }
+  }
+};
+
+/**
+ * Install is the first event which occurs when a new or updated ServiceWorker
+ * is registered.
+ */
+self.addEventListener("install", (event: ExtendableEvent) => {
+  swLog.V(`installing ${process.env.SW_ID}`);
+
+  if (!SERVICE_WORKER_ENABLED) {
+    return;
+  }
+
+  event.waitUntil(
+    (async () => {
+      // By default, the browser will only let one version of our ServiceWorker
+      // be active, and will wait to activate a new one until the old one isn't
+      // controlling any pages.  This call skips that so the new one can take
+      // over ASAP.
+      await swSelf.skipWaiting();
+
+      // Pre-populate our cache with all the resources we need to operate offline.
+      await prePopulateCaches();
+
+      swLog.I(`installed ${process.env.SW_ID}`);
```

Comment:
may be worth extracting `process.env.SW_ID` to a variable in this file named something like `BUILD_ID` w/ documentation as to the conditions under which it changes.